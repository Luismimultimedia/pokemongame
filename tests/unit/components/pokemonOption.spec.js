import { shallowMount } from "@vue/test-utils";
import PokemonOptions from "@/components/PokemonOption";
import { pokemons } from "../mocks/pokemons.mock";

describe('Test del componente pokemonOption', () => {

    let wrapper

    beforeEach(() => {
        wrapper = shallowMount(PokemonOptions, {
            props: {
                pokemons: pokemons
            }
        })
    })

    test('Debe de hacer match en el snapshot', () => {
        console.log(wrapper.html());
        expect(wrapper.html()).toMatchSnapshot()
    });

    test('Deben de renderizarse 4 elementos', () => {
        const liTags = wrapper.findAll('li')
        expect(liTags.length).toBe(4)
    });

    test('Debe de emitir "selection" con sus respectivos valores al hacer click', () => {
        const [l1, l2, l3, l4] = wrapper.findAll('li')
        l1.trigger('click')
        l2.trigger('click')
        l3.trigger('click')
        l4.trigger('click')

        expect(wrapper.emitted('selection')[0]).toStrictEqual([{ "id": 151, "name": 'mew' }])
        expect(wrapper.emitted('selection')[1]).toStrictEqual([{ "id": 5, "name": 'pikachu' }])
        expect(wrapper.emitted('selection')[2]).toStrictEqual([{ "id": 245, "name": 'tyranitar' }])
        expect(wrapper.emitted('selection')[3]).toStrictEqual([{ "id": 2, "name": 'charmander' }])
    });
});