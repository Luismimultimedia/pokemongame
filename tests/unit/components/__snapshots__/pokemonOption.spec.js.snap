// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test del componente pokemonOption Debe de hacer match en el snapshot 1`] = `
<div class="options-container">
  <ul>
    <li class="fade-in">mew</li>
    <li class="fade-in">pikachu</li>
    <li class="fade-in">tyranitar</li>
    <li class="fade-in">charmander</li>
  </ul>
</div>
`;
